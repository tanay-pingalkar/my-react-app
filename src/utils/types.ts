export type counterAction = {
  type: "add" | "sub";
};

export interface state {
  counter: number;
}
