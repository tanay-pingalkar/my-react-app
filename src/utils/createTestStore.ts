import { createStore } from "redux";
import { allReducer } from "../redux/combine";

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const createTestStore = (): any => {
  return createStore(allReducer);
};
