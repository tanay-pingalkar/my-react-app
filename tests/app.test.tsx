import {
  fireEvent,
  render,
  screen,
  act,
  cleanup,
} from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import React from "react";
import App from "../src/app";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { createTestStore } from "../src/utils/createTestStore";

describe("gonna test app", () => {
  let store;
  beforeEach(() => {
    store = createTestStore();
  });
  afterEach(() => {
    jest.clearAllMocks();
    cleanup();
  });
  it("should match snapshot", () => {
    const app = renderer
      .create(
        <MemoryRouter>
          <Provider store={store}>
            <App></App>
          </Provider>
        </MemoryRouter>,
      )
      .toJSON();
    expect(app).toMatchSnapshot();
  });
  it("should render app properly", async () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <App></App>
        </Provider>
      </MemoryRouter>,
    );
    const text = await screen.findByTestId("counter");
    const add = await screen.findByText("+");
    const sub = await screen.findByText("-");
    expect(text).toHaveTextContent("0");
    act(() => {
      fireEvent.click(add);
    });
    expect(text).toHaveTextContent("1");
    act(() => {
      fireEvent.click(sub);
    });
    act(() => {
      fireEvent.click(sub);
    });
    expect(text).toHaveTextContent("-1");
  });
});
