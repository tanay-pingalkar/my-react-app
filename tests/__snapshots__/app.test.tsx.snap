// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gonna test app should match snapshot 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    -
  </button>
  <h1
    data-testid="counter"
  >
    0
  </h1>
  <button
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
